### ISSUE TITLE
Invalid Form Body: BUTTON_COMPONENT_INVALID_EMOJI

### ISSUE THREAD

**Issue by 1Ckpwee** [bwmarrin/discordgo#1479]
I'm currently facing a peculiar issue where, without any changes to the code, I suddenly encounter a '400 (Invalid Emoji)' error. I can confirm that this code did not produce any errors during previous testing.
Here are the relevant log entries:
```bash
2024/01/09 18:58:41 API REQUEST POST :: https://discord.com/api/v9/interactions/xxx/xxx/callback
2024/01/09 18:58:41 API REQUEST PAYLOAD :: [{"type":4,"data":{"tts":false,"content":"xxxx","components":[{"components":[{"label":"xxx","style":5,"disabled":false,"emoji":{},"url":"https://discord.com/channels/xxx/xxx","type":2}],"type":1}],"embeds":null,"flags":64}}]
...
2024/01/09 18:58:42 API RESPONSE STATUS :: 400 Bad Request
2024/01/09 18:58:42 API RESPONSE BODY :: [{"message": "Invalid Form Body", "code": 50035, "errors": {"data": {"components": {"0": {"components": {"0": {"emoji": {"name": {"_errors": [{"code": "BUTTON_COMPONENT_INVALID_EMOJI", "message": "Invalid emoji"}]}}}}}}}}}]
```
Here is the relevant code:
```go
s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse{
    Type: discordgo.InteractionResponseChannelMessageWithSource,
    Data: &discordgo.InteractionResponseData{
        Flags:   discordgo.MessageFlagsEphemeral,
        Content: msg,
        Components: []discordgo.MessageComponent{
            discordgo.ActionsRow{
                Components: []discordgo.MessageComponent{
                    discordgo.Button{
                        Label: "xxx",
                        Style: discordgo.LinkButton,
                        URL:   channelLink,
                    },
                },
            },
        },
    },
})
```

I would appreciate any insights or suggestions to help resolve this issue. Thank you.

**Comment by 1Ckpwee** [comment]
As far as I know, the API documentation(https://discord.com/developers/docs/interactions/message-components#buttons) mentions that the 'emoji' field in the 'Button' is an optional field. 
![image](https://github.com/bwmarrin/discordgo/assets/51103782/3a4ebc37-e2fd-449f-a334-39ae2b0d525c)

**Comment by FedorLap2006** [comment]
Correct, it should be optional. We've already merged fixes for v0.24-v0.27 versions. So please update.

P.S. You'll have to use `@branch` revisions for now, releases coming soon.
